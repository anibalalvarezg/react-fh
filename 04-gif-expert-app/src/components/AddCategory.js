import React, { useState } from 'react';

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Hello World');

    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    };

    const handleSubmit = (event) => {
        event.preventDefault(); 
        console.log('Submit done!');
    };

    return (
        <form onSubmit={ handleSubmit }>
            <h1>{ inputValue }</h1>
            <input
                type="text"
                value={ inputValue } 
                onChange={ handleInputChange } 
            />
        </form>
    );
}
