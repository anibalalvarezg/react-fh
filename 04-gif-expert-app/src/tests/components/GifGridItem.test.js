import React from 'react'
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';
describe('Tests GifGridItem', () => {

    const title = 'A title'; 
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    test('should shows component correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should hava a paragrah with a title', () => {
        const p = wrapper.find('p'); 
        expect( p.text().trim() ).toBe( title );
    });
    
    test('should has a image equal to the url and the alt from prop', () => {
        const img = wrapper.find('img');
        console.log(img.props());
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    }); 

    test('should has animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__fadeIn') ).not.toBe( false );
    });
    
    
});