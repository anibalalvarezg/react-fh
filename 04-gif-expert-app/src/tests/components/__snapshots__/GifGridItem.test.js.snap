// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests GifGridItem should shows component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
